Randomizer: Fix Biggoron checks

lilDavid opened the pull request:
This fixes Biggoron's text being dependent on the BGS flag in rando. This fixes the follwing issues:
 - #1305
 - #739
 - https://github.com/HarbourMasters/rando-issue-tracker/issues/136 was already fixed, but now the textbox that gives you the sword doesn't appear until after you trade the Claim Check, rendering the original fix redundant.

lilDavid committed:
Make Biggoron ignore whether you have BGS in rando

aMannus commented:
Tested and confirmed working correctly in both vanilla and rando saves with this PR. :)

Cannot show him the claim check for multiple rewards either.

aMannus commented on soh/src/overlays/actors/ovl_En_Go2/z_en_go2.c:
```diff
@@ -1063,24 +1059,28 @@ void EnGo2_BiggoronSetTextId(EnGo2* this, GlobalContext* globalCtx, Player* play
 
     if ((this->actor.params & 0x1F) == GORON_DMT_BIGGORON) {
-        if (gSaveContext.bgsFlag) {
+        if ((!gSaveContext.n64ddFlag && gSaveContext.bgsFlag)) {
```
Very small nitpick. The other conditionals in this file have these 2 the other way around.

lilDavid replied:
`EnGo2_GetStateGoronDmtBiggoron` has the conditions in this order, so I fixed this by swapping the other conditionals instead of this one. That way, all five places that check both of these flags do it in the same order.

aMannus replied:
Fine with me 👍

lilDavid committed:
Always check n64ddFlag before bgsFlag
that order is consistent with the existing condition in EnGo2_GetStateGoronDmtBiggoron

briaguya0 commented on soh/src/overlays/actors/ovl_En_Go2/z_en_go2.c:
```diff
@@ -622,10 +622,6 @@ s16 EnGo2_GetStateGoronDmtBiggoron(GlobalContext* globalCtx, EnGo2* this) {
                 }
                 
                 if(gSaveContext.n64ddFlag) {
-                    if (INV_CONTENT(ITEM_CLAIM_CHECK) != ITEM_CLAIM_CHECK) {
-                        return 0;
-                    }
```
so is the idea we just never hit this without the claim check now?

lilDavid replied:
That's right. If you don't have the claim check, the text box that gives it to you doesn't appear anymore, so we shouldn't hit this.

briaguya0 reviewed (approved):
looking good, left a comment with a question then i'll hit merge!
